plugins/outputs/microsoft_fabric/microsoft_fabric.go

```diff
@@ -0,0 +1,94 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package microsoft_fabric
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/adx"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type fabric interface {
+	Connect() error
+	Write(metrics []telegraf.Metric) error
+	Close() error
+}
+
+type MicrosoftFabric struct {
+	ConnectionString string          `toml:"connection_string"`
+	Timeout          config.Duration `toml:"timeout"`
+	Log              telegraf.Logger `toml:"-"`
+
+	output fabric
+}
+
+func (*MicrosoftFabric) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *MicrosoftFabric) Init() error {
+	// Check input parameters
+	if m.ConnectionString == "" {
+		return errors.New("endpoint must not be empty")
+	}
+
+	// Initialize the output fabric dependent on the type
+	switch {
+	case isEventstreamEndpoint(m.ConnectionString):
+		m.Log.Debug("Detected EventStream endpoint...")
+		eventstream := &eventstream{
+			connectionString: m.ConnectionString,
+			timeout:          m.Timeout,
+			log:              m.Log,
+		}
+		if err := eventstream.init(); err != nil {
+			return fmt.Errorf("initializing EventStream output failed: %w", err)
+		}
+		m.output = eventstream
+	case isEventhouseEndpoint(strings.ToLower(m.ConnectionString)):
+		m.Log.Debug("Detected EventHouse endpoint...")
+		eventhouse := &eventhouse{
+			connectionString: m.ConnectionString,
+			Config: adx.Config{
+				Timeout: m.Timeout,
+			},
+			log: m.Log,
+		}
+		if err := eventhouse.init(); err != nil {
+			return fmt.Errorf("initializing EventHouse output failed: %w", err)
+		}
+		m.output = eventhouse
+	default:
+		return errors.New("invalid connection string")
+	}
+	return nil
+}
+
+func (m *MicrosoftFabric) Close() error {
+	return m.output.Close()
+}
+
+func (m *MicrosoftFabric) Connect() error {
+	return m.output.Connect()
+}
+
+func (m *MicrosoftFabric) Write(metrics []telegraf.Metric) error {
+	return m.output.Write(metrics)
+}
```

Comment:
Nil Pointer Risk
```suggestion
func (m *MicrosoftFabric) Close() error {
	if m.output == nil {
		return nil
	}
	return m.output.Close()
}

func (m *MicrosoftFabric) Connect() error {
	if m.output == nil {
		return errors.New("plugin not properly initialized")
	}
	return m.output.Connect()
}

func (m *MicrosoftFabric) Write(metrics []telegraf.Metric) error {
	if m.output == nil {
		return errors.New("plugin not properly initialized")
	}
	return m.output.Write(metrics)
}
```

Reply:
Hi @skartikey, this was removed as per review comment by @srebhan  here https://github.com/influxdata/telegraf/pull/16827/commits/360ada32e0017a02918f400bc84c741d7de84666#r2107663894

Since this will never be the case